elytron/src/main/java/org/wildfly/extension/elytron/PrincipalDecoderDefinitions.java

```diff
@@ -247,30 +258,48 @@ public void registerAttributes(ManagementResourceRegistration resourceRegistrati
 
     private static class PrincipalDecoderAddHandler extends BaseAddHandler {
 
+        private static final Set<RuntimeCapability> CAPABILITIES;
+
+        static {
+            Set<RuntimeCapability> capabilities = new HashSet<>(2);
+            capabilities.add(PRINCIPAL_DECODER_RUNTIME_CAPABILITY);
+            capabilities.add(PRINCIPAL_TRANSFORMER_RUNTIME_CAPABILITY);
+            CAPABILITIES = capabilities;
+        }
 
         private PrincipalDecoderAddHandler(AttributeDefinition ... attributes) {
-            super(PRINCIPAL_DECODER_RUNTIME_CAPABILITY, attributes);
+            super(CAPABILITIES, attributes);
```

Comment:
This is ok but we should rework BaseAddHandler to get rid of the need to pass capabilities to the add handler constructor. AFAICT the only reason BaseAddHandler exists is for this:

```
runtimeCapabilities.forEach(r -> context.registerAdditionalCapabilityRequirement(ELYTRON_CAPABILITY, r.isDynamicallyNamed() ? r.getDynamicName(pathValue) : r.getName(), null));
```

That would not be necessary if the various capabilities simply did

```
static final RuntimeCapability<Void> PRINCIPAL_DECODER_RUNTIME_CAPABILITY =  RuntimeCapability
        .Builder.of(PRINCIPAL_DECODER_CAPABILITY, true, PrincipalDecoder.class)
           .addRequirements(ELYTRON_CAPABILITY)
        .build();
```